src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -651,7 +651,9 @@ int wm_vuldet_update_agent_cpes(agent_software *agent, sqlite3 *db) {
         const char *version = (const char *) sqlite3_column_text(stmt, 14);
         const char *arch = (const char *) sqlite3_column_text(stmt, 15);
 
-        snprintf(buffer, OS_SIZE_6144, vu_queries[VU_SYSC_UPDATE_CPE],
+        // If the vendor or package string contains single quotation characters
+        // sqlite3_snprintf will escape them with double single quotes to avoid breaking the SQL query
+        sqlite3_snprintf(OS_SIZE_6144, buffer, vu_queries[VU_SYSC_UPDATE_CPE],
```

Comment:
The usage of this API resolves the issue of single quotes in the query.
https://www.sqlite.org/c3ref/mprintf.html
https://www.sqlite.org/printf.html#percentq